### PR TITLE
BVTCK-89 Improved java.time support

### PR DESCRIPTION
**Pull request by gsmet**
See https://github.com/beanvalidation/beanvalidation-api/pull/70 for a general presentation.

See https://github.com/hibernate/hibernate-validator/pull/576 for the HV changes.

Note: the references to the spec have to be updated once the spec change is done.

### CODE REVIEW

**Comment by gsmet** [comment]
Jenkins, restest this please

File: tests/src/main/java/org/hibernate/beanvalidation/tck/tests/xmlconfiguration/ClockProviderSpecifiedInValidationXmlNoDefaultConstructorTest.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
+/*
+* JBoss, Home of Professional Open Source
+* Copyright 2009, Red Hat, Inc. and/or its affiliates, and individual contributors
```
Comment (gunnarmorling) [inline tests/src/main/java/org/hibernate/beanvalidation/tck/tests/xmlconfiguration/ClockProviderSpecifiedInValidationXmlNoDefaultConstructorTest.java]: I'll change it to 2016. I've also filed https://hibernate.atlassian.net/browse/BVTCK-92 to get rid of this lengthy header altogether.

File: tests/src/main/java/org/hibernate/beanvalidation/tck/tests/xmlconfiguration/ClockProviderSpecifiedInValidationXmlNoDefaultConstructorTest.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+import org.jboss.test.audit.annotations.SpecVersion;
+import org.testng.annotations.Test;
+
+import org.hibernate.beanvalidation.tck.util.TestUtil;
+import org.hibernate.beanvalidation.tck.util.shrinkwrap.WebArchiveBuilder;
+
+import static org.testng.Assert.fail;
+
+/**
+ * @author Hardy Ferentschik
```
Comment (gunnarmorling) [inline tests/src/main/java/org/hibernate/beanvalidation/tck/tests/xmlconfiguration/ClockProviderSpecifiedInValidationXmlNoDefaultConstructorTest.java]: I'll change it to your name.

**Comment by gunnarmorling** [comment]
Rebased, applied and added one more commit for some clean-up. Thanks, @gsmet!

**Pull request closed without merging**